core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1800,6 +1800,16 @@ package object config {
       .booleanConf
       .createWithDefault(true)
 
+  private[spark] val SHUFFLE_FAST_FAIL_ON_FETCH_FAILURE_ENABLED =
+    ConfigBuilder("spark.shuffle.fastFailOnFetchFailure")
+      .doc("If enabled FailureFetchResult will be added to the head of the result queue in " +
+        "ShuffleBlockFetcherIterator and fail the task as soon as the failure result is " +
+        "consumed. Otherwise, will have to process all the successfully fetched blocks in " +
```

Comment:
```suggestion
        "consumed. Otherwise, the iterator must first process all the successfully fetched blocks that are ahead in the queue before failing the task " +
```